libs/ledger-live-common/src/families/icon/js-estimateMaxSpendable.ts

```diff
@@ -0,0 +1,24 @@
+import { BigNumber } from "bignumber.js";
+import type { AccountLike, Account } from "@ledgerhq/types-live";
+import { getMainAccount } from "../../account";
+import type { IconAccount, Transaction } from "./types";
+import { TRANSACTION_FEE } from "./constants";
+
+/**
+ * Returns the maximum possible amount for transaction
+ *
+ * @param {Object} param - the account, parentAccount and transaction
+ */
+const estimateMaxSpendable = async ({
+  account,
+  parentAccount,
+}: {
+  account: AccountLike;
+  parentAccount?: Account;
+  transaction?: Transaction;
+}): Promise<BigNumber> => {
+  const a = getMainAccount(account, parentAccount) as IconAccount;
+  return a.spendableBalance.minus(TRANSACTION_FEE);
```

Comment:
From my understanding of how fees work with Icon, this still seems unreliable.

Fees actually paid in the end can be different from `TRANSACTION_FEE`, correct?
If that is the case, then the estimate could be either too high (==> transaction will fail) or too low (==> dust left in the account)

Isn't there any way to know fees in advance, using icon's SDK maybe?

Reply:
> Isn't there any way to know fees in advance, using icon's SDK maybe? 

Yes we have other ways to know the fee but we need to build the exact amount value for the transfer transaction, but the problem is that once we get the fee this way and then the amount value to send max will be changed by the fee and dust left in the account

For example: I have 100 ICX
- Expected send Max: 100 ICX => Using SDK to get fee = 0,01 ICX with amount is 100 ICX
- Actual send max: 100 - 0,01 = 0.99 ICX the free will be change and dust left in the account

The logic to set a constant value for free we refer from main wallet on ICON (HANA wallet), so you can take a look here https://github.com/Hana-Technology/myiconwallet-react/blob/a0363491883d2e584e9b0d43a3cddfa5fda5f040/src/pages/Send.jsx#L38C64-L38C64

Reply:
@heirland Ok I understand better, thanks for the explanation :slightly_smiling_face: if exact fees are computed down the way by the SDK then it's OK.

Anyway, it often happens that fees are kindof always the same (as long as the network doesn't start to be used a lot...) and hardocding them works just fine. So in the end up to you, if you think those hardcoded fees work then I'm OK with it.